src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryPublisher.cs

```diff
@@ -66,5 +71,26 @@ private void ValidateEvent(EvaluationEvent evaluationEvent)
                 throw new ArgumentNullException(nameof(evaluationEvent.FeatureDefinition));
             }
         }
+
+        private static string ToString(AssignmentReason reason)
+        {
+            const string None = "None";
```

Comment:
Since we have implemented a ToString method, shall we add more detailed variant assignment reason? For example, AssignmentReason.None can be mapped to "No variant can be allocated".

Reply:
These are metric dimensions, I don't think we should be verbose.